frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx

```diff
@@ -65,28 +65,27 @@ export const generateClientKeySecret = (clientKey: string) => ({
   },
 });
 
-export const createKmsResources = (kms: KMSConfig, update = false, previousData?: any) => {
-  let tokenSecret: SecretKind;
-  if (kms.token) {
-    tokenSecret = {
-      apiVersion: SecretModel.apiVersion,
-      kind: SecretModel.kind,
-      metadata: {
-        name: KMSSecretName,
-        namespace: CEPH_STORAGE_NAMESPACE,
-      },
-      stringData: {
-        token: kms.token.value,
-      },
-    };
-  }
+export const createCommonKmsResources = (kms: KMSConfig) => {
+  const commonKmsResources: Promise<K8sResourceKind>[] = [];
 
-  const resources: Promise<K8sResourceKind>[] = [];
+  if (kms.caCert)
+    commonKmsResources.push(k8sCreate(SecretModel, kms.caCert, CEPH_STORAGE_NAMESPACE));
+  if (kms.clientCert)
+    commonKmsResources.push(k8sCreate(SecretModel, kms.clientCert, CEPH_STORAGE_NAMESPACE));
+  if (kms.clientKey)
+    commonKmsResources.push(k8sCreate(SecretModel, kms.clientKey, CEPH_STORAGE_NAMESPACE));
 
-  const parsedAddress = parseURL(kms.address.value);
+  return commonKmsResources;
+};
 
-  const configData: KMSConfigMap = {
-    KMS_PROVIDER: 'vault',
+export const createCsiKmsResources = (
+  kms: KMSConfig,
+  update: boolean,
+  clusterKMSresources?: Promise<K8sResourceKind>[],
```

Comment:
Why you are passing clusterKMSresources of Promise<K8sResourceKind>[], can we create a simple array of K8sResourceKind and you can receive or concat of Promise<K8sResourceKind>[] type.  No need to keep promise to all the places.